pkg/collector/corechecks/servicediscovery/module/comm_test.go

```diff
@@ -67,35 +63,88 @@ func TestIgnoreCommsLengths(t *testing.T) {
 	}
 }
 
-// buildLongProc returns the path of a symbolic link to a binary file
-func buildLongProc(t *testing.T) string {
+// TestIgnoreKernelThread check that kernel thread is ignored
+func TestIgnoreKernelThread(t *testing.T) {
+	proc := &process.Process{
+		Pid: 1,
+	}
+	// Pid=1 is not kernel thread and should not be ignored
+	ignore := shouldIgnoreKernelThread(proc)
```

Comment:
the tests passes just because you have a processes with a pid `1`

Reply:
Yes, Pid=1 is init and is not kernel thread